components/atom/tooltip/src/index.js

```diff
@@ -0,0 +1,108 @@
+import React, {Component} from 'react'
+import PropTypes from 'prop-types'
+import {Tooltip} from 'reactstrap'
+import DOMElement from './customPropTypes/DOMElement'
+
+const BASE_CLASS = 'sui-AtomTooltip'
+
+const PLACEMENTS = {
+  TOP: 'top',
+  TOP_START: 'top-start',
+  TOP_END: 'top-end',
+  TOP_RIGHT: 'right',
+  RIGHT_START: 'right-start',
+  RIGHT_END: 'right-end',
+  BOTTOM: 'bottom',
+  BOTTOM_START: 'bottom-start',
+  BOTTOM_END: 'bottom-end',
+  LEFT: 'left',
+  LEFT_START: 'left-start',
+  LEFT_END: 'left-end'
+}
+
+class AtomTooltip extends Component {
+  state = {
+    isOpen: false
+  }
+
+  componentDidMount() {
+    document.addEventListener('click', this.handleClickOutsideElement)
+  }
+
+  componentWillUnmount() {
+    document.removeEventListener('click', this.handleClickOutsideElement)
+  }
+
+  handleClickOutsideElement = event => {
+    const {isOpen} = this.state
+    if (isOpen) {
+      const tooltipDom = document.querySelector(`.${BASE_CLASS}`)
```

Comment:
The thing is this element is added and removed from the DOM every time the value of `isOpen` change, so i don't think caching this value would work

Reply:
ref means react will store this value when he creates the DOM element. So no search step needed.
Plus, if you identify by class you should search by class too: https://jsperf.com/getelementsbyclassname-vs-queryselectorall/67

Reply:
As discussed in person w/ @davidbarna we don't have access to the DOM element on the inside because this component is internally using the `reactstrap` Tooltip component. So we cannot access to this DOM elements via `ref`

There's an issue opened related to this (https://github.com/reactstrap/reactstrap/issues/1082) that maybe someday will give us access to the DOM elements inside and will allow to perform this kind of operations in a more elegant way

Meanwhile there's no other way than selecting the DOM element with some of available methods. In this case, using the `getElementsByClassName` seems to be the most easy and performant way

Reply:
As discussed w/ @davidbarna there's no way of using `ref` to access the inner DOM element with the current implementation of `Tooltip`'s reactstrap
I've created this PR (https://github.com/reactstrap/reactstrap/pull/1090) to be able to use `ref` to access the DOM element containing the Tooltip _from the outside_
Meanwhile using `getElementsByClassName` to get this element seems to be the most straightforward way.